packages/Utils/src/DTK_KokkosHelpers.hpp

```diff
@@ -14,31 +14,36 @@
 
 #include <Kokkos_Core.hpp>
 
+#include <type_traits>
+
 namespace DataTransferKit
 {
 namespace KokkosHelpers
 {
 
 //! Compute the maximum of two values.
-template <class SC>
-KOKKOS_INLINE_FUNCTION SC max( const SC left, const SC right )
+template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
+KOKKOS_INLINE_FUNCTION T max( T a, T b )
 {
-    return ( left > right ) ? left : right;
+    return ( a > b ) ? a : b;
 }
 
 //! Compute the minimum of two values.
-template <class SC>
-KOKKOS_INLINE_FUNCTION SC min( const SC left, const SC right )
+template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
+KOKKOS_INLINE_FUNCTION T min( T a, T b )
 {
-    return ( left < right ) ? left : right;
+    return ( a < b ) ? a : b;
 }
 
 /**
  * Branchless sign function. Return 1 if @param x is greater than zero, 0 if
  * @param x is zero, and -1 if @param x is less than zero.
  */
-KOKKOS_INLINE_FUNCTION
-int sgn( int x ) { return ( x > 0 ) - ( x < 0 ); }
+template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
+KOKKOS_INLINE_FUNCTION T sgn( T x )
```

Comment:
Hold on.  What should be the return type here?

Reply:
I think it should be `int`

Reply:
Probably `int` as the common denominator.

Reply:
```diff
diff --git a/packages/Utils/src/DTK_KokkosHelpers.hpp b/packages/Utils/src/DTK_KokkosHelpers.hpp
index 3240f99..8ee5655 100644
--- a/packages/Utils/src/DTK_KokkosHelpers.hpp
+++ b/packages/Utils/src/DTK_KokkosHelpers.hpp
@@ -40,7 +40,7 @@ KOKKOS_INLINE_FUNCTION T min( T a, T b )
  * @param x is zero, and -1 if @param x is less than zero.
  */
 template <typename T, typename = std::enable_if<std::is_arithmetic<T>::value>>
-KOKKOS_INLINE_FUNCTION T sgn( T x )
+KOKKOS_INLINE_FUNCTION int sgn( T x )
 {
Enable KokkosHelpers::sgn() for arithmetic types
     return ( x > 0 ) - ( x < 0 );
 }
```
Amended this to previous commit and pushed force.